src/file/format/ini.rs

```diff
@@ -3,35 +3,45 @@ use std::error::Error;
 use ini::Ini;
 
 use crate::map::Map;
-use crate::value::{Value, ValueKind};
+use crate::value::{Table, Value, ValueKind};
 
 pub fn parse(
     uri: Option<&String>,
     text: &str,
 ) -> Result<Map<String, Value>, Box<dyn Error + Send + Sync>> {
-    let mut map: Map<String, Value> = Map::new();
-    let i = Ini::load_from_str(text)?;
-    for (sec, prop) in i.iter() {
-        match sec {
-            Some(sec) => {
-                let mut sec_map: Map<String, Value> = Map::new();
-                for (k, v) in prop.iter() {
-                    sec_map.insert(
-                        k.to_owned(),
-                        Value::new(uri, ValueKind::String(v.to_owned())),
-                    );
-                }
-                map.insert(sec.to_owned(), Value::new(uri, ValueKind::Table(sec_map)));
-            }
-            None => {
-                for (k, v) in prop.iter() {
-                    map.insert(
-                        k.to_owned(),
-                        Value::new(uri, ValueKind::String(v.to_owned())),
-                    );
-                }
-            }
-        }
+    let value = from_ini(uri, Ini::load_from_str(text)?);
+
+    match value.kind {
+        ValueKind::Table(map) => Ok(map),
+
+        _ => Ok(Map::new()),
+    }
+}
+
+fn from_ini(
+    uri: Option<&String>,
+    data: Ini,
+) -> Value {
+    let mut map = Map::new();
+
+    let mut sections: Map<Option<&str>, Table> = data.into_iter().map(|(section, props)| {(
+        section,
+        props.iter().map(|(k, v)| {(
+            k.to_owned(),
+            Value::new(uri, ValueKind::String(v.to_owned())),
+        )}).collect()
```

Comment:
This is really unreadable.

How about
```suggestion
        props.iter().map(|(k, v)| {
            let key = k.to_owned();
            let value = Value::new(uri, ValueKind::String(v.to_owned());
            (key, value)
        }).collect()
```

instead of the rather hard to parse tuple inside the block?

Reply:
Yeah that's much better, thanks for the suggestion! :)